### PR TITLE
fix: empty product returns identity

### PR DESCRIPTION
**Pull request by orlp**
Closes https://github.com/pola-rs/polars/issues/8184 and https://github.com/pola-rs/polars/issues/8500.

### CODE REVIEW

File: py-polars/tests/unit/series/test_series.py [hunk 1]
```diff
@@ -2296,10 +2296,19 @@ def test_product() -> None:
     assert out == 6
     a = pl.Series("a", [1, 2, None])
     out = a.product()
-    assert out is None
+    assert out == 2
```
Comment (stinodego) [inline py-polars/tests/unit/series/test_series.py:2299]: Ignoring `null` values by default IS a breaking change. I'd suggest leaving that for a different PR, and making this one just add the identity for empty Series.

Comment (orlp) [reply py-polars/tests/unit/series/test_series.py:2299]: What about `sum`? It currently doesn't have such an option and already ignores nulls. I'm just making it consistent with `sum`.

Comment (stinodego) [reply py-polars/tests/unit/series/test_series.py:2299]: I get that, but it would be a breaking change. Ignoring nulls is not a bug.

Comment (ritchie46) [reply py-polars/tests/unit/series/test_series.py:2299]: I would say our product implementation is a bug. It didn't adhere to our set semantics/rules. Given that 0.19.0 will be patched and yanked I will include this one.

**Pull request merged**